tests/vmi_configuration_test.go

```diff
@@ -2508,6 +2508,20 @@ var _ = Describe("[sig-compute]Configurations", func() {
 				pinnedCPUsList, err := hw_utils.ParseCPUSetLine(output, 100)
 				Expect(err).ToNot(HaveOccurred())
 
+				output, err = tests.ListCgroupThreads(readyPod)
+				Expect(err).ToNot(HaveOccurred())
+				pids := strings.Split(output, "\n")
+
+				By("Expecting only vcpu threads on root of pod cgroup")
+				for _, pid := range pids {
+					output, err = tests.GetProcessName(readyPod, pid)
+					if err != nil {
```

Comment:
I think if you are checking the vcpu threads here, you should not ignore the error.

Reply:
@vasiliy-ul 

The "cat" process (used to read the list of PIDs in the cgroup) shows up, therefore we can't fail on error here:

```func ListCgroupThreads(pod *k8sv1.Pod) (output string, err error) {
        virtClient, err := kubecli.GetKubevirtClient()
        if err != nil {
                return
        }

        output, err = ExecuteCommandOnPod(
                virtClient,
                pod,
                "compute",
                []string{"cat", "/sys/fs/cgroup/cpuset/tasks"},
        )

        if err == nil {
                // Cgroup V1
                return
        }
        output, err = ExecuteCommandOnPod(
                virtClient,
                pod,
                "compute",
                []string{"cat", "/sys/fs/cgroup/cgroup.threads"},
        )
        return
}```

    {"component":"tests","level":"info","msg":"GetProcessName error failed executing command on pod: command terminated with exit code 1: stderr cat: /proc/120/comm: No such file or directory\n: stdout: ","pos":"vmi_configuration_test.go:2544","timestamp":"2022-10-14T19:42:49.728996Z"}

Reply:
Hm... not sure, I follow. Do you mean `tests.GetProcessName` fails with certain pids? That does not sound right to me. If that is the case, how can you be sure that the test actually 'tests' something further?

Reply:
Yes:

1) Retrieve list of TIDs in the cgroup (that will include the "cat" command used to read the list of processes from 
 /sys/cgroup/.../pod-cgroup/threads file).

2) Attempt to retrieve the process names of each TID from 1). This will fail since "cat" process has already exited.

The test ensures that there are no long running processes on the main pod cgroup (those that are in the threads file) except vCPU threads. So the test is still valid even though that error is ignored (and should be able to fix this issue and check that error once "k exec" processes can start running on the housekeeping cgroup).

Reply:
Then you should ignore only that specific error caused by `cat`. If you ignore 'any' possible error, the test will pass even if something is really broken.

Reply:
Vasily,

How can i ignore that particular error from cat? Only have the list of PIDs, without knowing
process names.

Well, again, the test will cause an error if a long lived process other than vCPU processes is on the housekeeping cgroup, which is what is desired.

Reply:
You have the list of pids, and you know that it contains the vCPU pids + one junk pid from cat. You can set the expectations so that `GetProcessName` fails **once** but shall succeed for the vCPU pids.

With current implementation, if for whatever reason `GetProcessName` fails for *all* pids, the test will still pass. That contradicts with what you claim:  `By("Expecting only vcpu threads on root of pod cgroup")`

Reply:
Fixed.